jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -2485,23 +2483,15 @@ ProtectionDomain getProtectionDomainInternal() {
 	if (result != null) {
 		return result;
 	}
-	if (allPermissionsPD == null) {
-		allocateAllPermissionsPD();
-	}
-	return allPermissionsPD;
+	return AllPermissionsPDHolder.allPermissionsPD;
 }
 
-private void allocateAllPermissionsPD() {
-	/* Synchronization to ensure safe initialization and safe publication of
-	 * allPermissionsPD. This addresses the locking contention on allPermissionsPD
-	 * in getProtectionDomain().
-	 */
-	synchronized(this) {
-		if (allPermissionsPD == null) {
-			Permissions collection = new Permissions();
-			collection.add(SecurityConstants.ALL_PERMISSION);
-			allPermissionsPD = new ProtectionDomain(null, collection);
-		}
+private static class AllPermissionsPDHolder {
+	private static final ProtectionDomain allPermissionsPD;
```

Comment:
This class should be `final`.
The field should not be `private` (else an accessor method is generated to read it).
This class should have an explicit (private) constructor; otherwise a warning is issued for newer Java verions.

Reply:
Updated.